boot/seal.go

```diff
@@ -819,53 +804,6 @@ func IsResealNeeded(pbc PredictableBootChains, bootChainsFile string, expectRese
 	return true, c + 1, nil
 }
 
-func postFactoryResetCleanupSecboot() error {
-	// we are inspecting a key which was generated during factory reset, in
-	// the simplest case the sealed key generated previously used the main
-	// handles, while the current key uses alt handles, hence we need to
-	// release the main handles corresponding to the old key
-	handles := []uint32{secboot.RunObjectPCRPolicyCounterHandle, secboot.FallbackObjectPCRPolicyCounterHandle}
-	usesAlt, err := UsesAltPCRHandles()
-	if err != nil {
-		return fmt.Errorf("cannot inspect fallback key: %v", err)
-	}
-	if !usesAlt {
-		// current fallback key using the main handles, which is
-		// possible of there were subsequent factory reset steps,
-		// release the alt handles associated with the old key
-		handles = []uint32{secboot.AltRunObjectPCRPolicyCounterHandle, secboot.AltFallbackObjectPCRPolicyCounterHandle}
-	}
-	return secbootReleasePCRResourceHandles(handles...)
-}
-
-func postFactoryResetCleanup() error {
-	hasHook, err := HasFDESetupHook(nil)
-	if err != nil {
-		return fmt.Errorf("cannot check for fde-setup hook %v", err)
-	}
```

Comment:
this seems to have had the check that the other place has a FIXME about?  can you explain why you removed this bit and moved into deleteOldKeys, which now has a FIXME?

Reply:
There are 2 other FIXMEs. But looking at them, I think there is not much to do there.
 * There is nothing wrong in looking at old key to see if they are tpm2 platform. We do not need the context.
 * If we do not find any old tpm2 keys, well that can also be due to a clean up being run again. So there is nothing to complain about.

However, when parsing key files, we do need a hint for the very old keys. I will add that. But that is just a hint.